src/pylibsshext/session.pyx

```diff
@@ -302,6 +320,54 @@ cdef class Session(object):
                                       self._fingerprint_py,
                                       know_host_msg)
 
+    def authenticate_specific_pubkey(
```

Comment:
I tested after applying the changes here https://gist.github.com/ganeshrn/46eafd00e1f91ba4d5a05d7b1a441ab5 on top of your PR and it is working fine for me now

Reply:
Applied, let's see what happens.